services/dask-sidecar/src/simcore_service_dask_sidecar/computational_sidecar/models.py

```diff
@@ -1,29 +1,59 @@
-from typing import Dict, List
+from typing import Optional
 
 from packaging import version
-from pydantic import BaseModel, ByteSize, Field
+from pydantic import BaseModel, ByteSize, Field, validator
 
 LEGACY_INTEGRATION_VERSION = version.Version("0")
 
 
 class ContainerHostConfig(BaseModel):
-    binds: List[str] = Field(
+    # NOTE: https://github.com/ITISFoundation/osparc-simcore/issues/3506
+    # Be careful! --priviledged, --pid=host --cap-add XXX should never be usable here!!
+    # at the moment they are not part of the possible configuration but if they were
+    # to, ensure they are properly validated
+
+    binds: list[str] = Field(
         ..., alias="Binds", description="A list of volume bindings for this container"
     )
     init: bool = Field(
-        True,
+        default=True,
         alias="Init",
         description="Run an init inside the container that forwards signals and reaps processes. This field is omitted if empty, and the default (as configured on the daemon) is used",
     )
     memory: ByteSize = Field(..., alias="Memory", description="Memory limit in bytes")
+    memory_swap: Optional[ByteSize] = Field(
+        default=None,
+        alias="MemorySwap",
+        description="Total memory limit (memory + swap). Set as -1 to enable unlimited swap.",
+    )
     nano_cpus: int = Field(
         ..., alias="NanoCPUs", description="CPU quota in units of 10-9 CPUs"
     )
 
+    @validator("memory_swap", pre=True, always=True)
+    @classmethod
+    def ensure_no_memory_swap_means_no_swap(cls, v, values):
+        if v is None:
+            # if not set it will be the same value as memory to ensure swap is disabled
+            return values["memory"]
+        return v
+
+    @validator("memory_swap")
+    @classmethod
+    def ensure_memory_swap_is_either_disabled_or_greater_or_equal_to_memory(
+        cls, v, values
+    ):
+        if v <= -1:
```

Comment:
very minor: would it be straight forward to raise an exception or complain if `v < -1` and only accept `v = -1`? I'd like this

Reply:
It is of course possible. Although I don't really see the point? 
anyway I see the docker engine complains if it I try to run with -10 so I can do the same.